apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -1292,12 +1292,12 @@ static void tc_pthread_pthread_cond_signal_wait(void)
 	ret_chk = pthread_create(&pthread_waiter, NULL, thread_cond_signal, NULL);
 	TC_ASSERT_EQ("pthread_create", ret_chk, OK);
 
-	/* if g_cond_sig_val is not matched with VAL_ONE, then TC fails. but we will not use g_mutex anymore, so destroy it */
-	TC_ASSERT_EQ_CLEANUP("pthread_mutex_lock", g_cond_sig_val, VAL_ONE, pthread_mutex_destroy(&g_mutex));
-
 	ret_chk = pthread_mutex_unlock(&g_mutex);
 	TC_ASSERT_EQ("pthread_mutex_unlock", ret_chk, OK);
 
+	/* if g_cond_sig_val is not matched with VAL_ONE, then TC fails. but we will not use g_mutex anymore, so destroy it */
+	TC_ASSERT_EQ_CLEANUP("pthread_mutex_lock", g_cond_sig_val, VAL_ONE, pthread_mutex_destroy(&g_mutex));
```

Comment:
It is wrong modification.
Please put in original place, and add `pthread_mutex_unlock` in assert macro.